src/odinapi/database/level2db.py

```diff
@@ -437,6 +443,25 @@ def collapse_products(products):
     return list(prods.values())
 
 
+def get_valid_collapsed_products(products, limit):
+    """wraps around collapse_products and respecting a limit
+       to ensure that uncomplete products are not collapsed
+    """
+    next_min_scanid = get_next_min_scanid(products, limit)
+    collapsed_products = []
+    for scanid, scan in groupby(products, itemgetter('ScanID')):
+        if scanid == next_min_scanid:
+            continue
+        collapsed_products.extend(collapse_products(list(scan)))
+    return collapsed_products
+
+
+def get_next_min_scanid(products, limit):
```

Comment:
This is not valid because limit can be any number and set to the exact number of the products in the last scan to be returned and then the data will never be returned. The check must know if there were more products in the database that matched the criteria or not.

Also why `0`, it's a very strange return value, isn't `None` more clear?

Reply:
It also misses the third case. If we are investigating the last scan of all scans included in the query settings and lets say it has 50 documents but you've set the limit to 49. Then the check must raise and exception and then endpoint should return a `BadRequest` indicating that the limit was set so not all data could be returned.

Reply:
0 is reasonable, since that is the lowest theoretically possible scanid.

Reply:
why is `next_min_scanid=0` reasonable answer for the case of we got everything we needed there are no more scans to collect? I'd say `math.inf` may be reasonable, but never 0.

Reply:
will return none. added an assert to be more clear under what condition the function is valid,
in practise len(products) can not be greater than limit  (if the input come from mongodb)